### DISCUSSION TITLE
How to create layout with multiple columns?

### DISCUSSION THREAD

**Discussion by jordan-bravo** [zellij-org/zellij#3075] [Q&A] [answered]
I'm reading through the documentation but I'm having trouble figuring out how to accomplish the following layouts, 

For example three panes side-by-side: 

```
----------------------------------------------
|             |               |              |
|             |               |              |
|             |               |              |
|             |               |              |
|             |               |              |
|             |               |              |
|             |               |              |
|             |               |              |
----------------------------------------------
```

Or four panes side-by-side:

```
------------------------------------------------
|          |           |           |           |
|          |           |           |           |
|          |           |           |           |
|          |           |           |           |
|          |           |           |           |
|          |           |           |           |
|          |           |           |           |
|          |           |           |           |
------------------------------------------------
```

I have tried the following:
```kdl
layout {
    default_tab_template {
        pane borderless=true split_direction="vertical"
    }
}
```

Any help is much appreciated.

**Comment by Lockszmith-GH** [comment]
Instead of providing you with a solution, may I suggest you utilize the built-in auto-saving option and some trial and error.

using Ctrl+P N and resizing with Alt <Arrows>, get the layout you want, and then check the zellij cache:

```bash
less $(find ~/.cache/zellij -name 'session-layout.kdl')
```

hope this helps.

**Comment by jordan-bravo** [comment]
I appreciate the guidance.  

Unfortunately, I can't rearrange the panes into more than two columns. without also having half the window split (see below).

When I try `vertical`:
```
-----------------------------------
|                 |               |
|                 |               |
|                 |               |
|                 |---------------|
|                 |               |
|                 |               |
|                 |               |
-----------------------------------
```

When I try `horizontal`:
```
-----------------------------------
|                 |               |
|                 |               |
|                 |               |
|-----------------|---------------|
|                                 |
|                                 |
|                                 |
-----------------------------------
```

When I try `stacked`:
```
-----------------------------------
|                 |---------------|
|                 |               |
|                 |               |
|                 |               |
|                 |               |
|                 |               |
|                 |               |
-----------------------------------
```

Is it even possible for Zellij to have a 3+ columns-only layout?

**Reply by Lockszmith-GH** [reply]
the split is by default 50% of the pane being split, so for a 3 column, you'll need to specify that the first split is 66%, and the next will be 50% (that's when coding it)

For me the following sequence of keyboard shortcuts worked out:
`Ctrl-P R` `Ctrl-N` `<-` `<-` `<-` `<-` `<Esc>` `Ctrl-P R`

**Reply by Lockszmith-GH** [reply, marked as answer]
Looking at the layout this is what I extracted as relevant:

```kdl
layout {
    tab name="Tab #1" hide_floating_panes=true {
        pane size=1 borderless=true {
            plugin location="zellij:tab-bar"
        }
        pane split_direction="vertical" {
            pane size="34%"
            pane size="33%"
            pane size="33%"
        }
        pane size=2 borderless=true {
            plugin location="zellij:status-bar"
        }
    }
}
```

**Comment by jordan-bravo** [comment]
Thank you so much, @Lockszmith-GH , this was the missing piece of the puzzle, specifically the "R" key in Pane Mode.

I think what tripped me up was that I was reading the documentation on zellij.dev and it seems to be missing the page with default keybindings, but I was able to find it on GitHub:  https://github.com/zellij-org/zellij/blob/main/zellij-utils/assets/config/default.kdl

When I have time, I'll make a pull request to add the documentation of default keybindings to the zellij.dev documentation site (if someone else doesn't get to it first).

**Reply by Lockszmith-GH** [reply]
I recommend you get familiar with dumping the configuration, that's how I learned those key-bindings
